web/moduly/program.php

```diff
@@ -26,38 +26,46 @@
   float: right;
   display: block;
   text-transform: uppercase;
-  padding: 8px 20px;
+  padding: 6px 20px;
   border: solid 1px #444;
   background-color: #d13f3f;
   color: #fff;
   font-weight: bold;
-  margin: -8px 0px 0px 24px;
-  font-size: 1.2em;
+  margin: -8px 0px 0px 16px;
+  font-size: 14px;
   border-radius: 6px;
 }
 </style>
 
-<?php if(!$this->param('osobni')) { ?>
-  <div id="programSkryvaniLinii_ovladani" class="programSkryvaniLinii_ovladani">
-    <span class="programSkryvaniLinii_popisek">Filtrovat linie: </span>
-  </div>
-<?php } ?>
-
-<?php if($u) { ?>
-  <a class="muj-program" target="_blank" href="programKTisku">k tisku</a>
-  <?php if($this->param('osobni')) { ?>
-    <a class="muj-program" href="program">celkový program</a>
-  <?php } else { ?>
-    <a class="muj-program" href="muj-program">můj program</a>
+<?php require __DIR__ . '/../soubory/program-nahled.html'; ?>
+
+<div class="programNahled_obalProgramu">
+
+  <?php if(!$this->param('osobni')) { ?>
+    <div id="programSkryvaniLinii_ovladani" class="programSkryvaniLinii_ovladani">
+      <span class="programSkryvaniLinii_popisek">Filtrovat linie: </span>
+    </div>
   <?php } ?>
-<?php } ?>
 
-<?=$legenda?>
+  <a class="muj-program" id="programNahled_externiPrepinac" href="#"></a>
+  <?php if($u) { ?>
+    <a class="muj-program" target="_blank" href="programKTisku">k tisku</a>
+    <?php if($this->param('osobni')) { ?>
+      <a class="muj-program" href="program">celkový program</a>
+    <?php } else { ?>
+      <a class="muj-program" href="muj-program">můj program</a>
+    <?php } ?>
+  <?php } ?>
+
+  <?=$legenda?>
+
+  <?php $program->tisk(); ?>
 
-<?php $program->tisk(); ?>
+</div>
 
 <script>
 programSkryvaniLinii($('table.program'), $('#programSkryvaniLinii_ovladani'));
+programNahled($('.programNahled_obalNahledu'), $('.programNahled_obalProgramu'), $('.programNahled_odkaz'), $('#programNahled_externiPrepinac'));
```

Comment:
Ten náhled může být tuším jen jeden, neuděláme proto z těhle specializovaných tříd raději IDčka?
`$('#programNahled_obalNahledu'), $('#programNahled_obalProgramu'), $('#programNahled_odkaz')`
Kdybych se díval jen do javascriptu, tak bych byl podle tříd přesvědčený, že se ty elementy na stránce opakují a je škoda tajit, že jsou unikátní.

Reply:
Moc se nevyznam v JS, takze nevim, jestli je na to nejaky standardni styl. Z jinych jazyku jsem se spis setkal s doporucenim predavat veci jako objekty nez jako stringove identifikatory (nejenom kvuli testovani).

Jquery pouziva na vsechno mnozinu elementu (objekt typu jeden element tam myslim ani neni), takze jsem vychazel z toho. Na druhou stranu, ze napriklad tech prepinacich tlacitek pak muze byt jedno nebo vice, podle toho co tam predas, by se dalo brat i za jistou vyhodu.

Reply:
Jestli tam může být některých těch prvků víc, tak je to OK.

Jinak jsem nenavrhoval stringové identifikátory, ale jeden jQuery objekt, zacílený přes ID

- `$('.foo')` = mnozina všech elementů s třídou `foo` (obalených do jQuery objektu)
-- jQuery je supr v tom, že zavolám jednou nějakou jQuery funkci na tuhle množinu a provede se nad všemi elementy v té množině
- `$('#foo')` = jeden element s ID `foo` (obalený do jQuery objektu)

Ale jak píšeš, tady je to zbytečné, když se ty elementy můžou opakovat.